### ISSUE TITLE
What is the suggested way to load the exported quantized model?

### ISSUE THREAD

**Issue by jasonlovescoding** [quic/aimet#88]
There are a .onnx file, a .pth file and a .decodings file after the export
Can a QuantizationSimModel be built from them?
Thanks!

**Comment by quic-ssiddego** [comment]
Hi @jasonlovescoding Yes, QuantizationSim model can be built from an exported quantized model. 
Could you please share more details on the specific use case you are exploring?

**Comment by flaviomb** [comment]
I'm having the same doubt.
I'm using the example described in the [api doc tutorial](https://quic.github.io/aimet-pages/AimetDocs/api_docs/torch_quantsim.html#code-examples). In this example, the `quantize_model` function has a command to save the quantized model:

```
 ### Export the model
    sim.export(path='./', filename_prefix='quantized_mnist', input_shape=input_shape)
```
This method return three files: 

```
quantized_mnist.encodings
quantized_mnist.onn
quantized_mnist.pth
```

Regarding this files, I have two specific questions:

1) How to restore my quantized model using three files?

2) It is possible to load the quantized model in a machine without the Qualcomm Aimet installed?


Here is my code for reproduction purposes:
```

import torch
from aimet_torch.examples import mnist_torch_model
# Quantization related import
from aimet_torch.quantsim import QuantizationSimModel
import torchvision

import torch.nn as nn
import torch.nn.functional as F
import torch.optim as optim

import torch.nn.functional as F

from pipelines import train, evaluate_model
n_epochs = 1
batch_size_train = 64
batch_size_test = 1000
learning_rate = 0.01
momentum = 0.5
log_interval = 10

train_loader = torch.utils.data.DataLoader(
  torchvision.datasets.MNIST('/home/flavio/git/qualcomm_aimet_tool/pytorch_example/files/', train=True, download=True,
                             transform=torchvision.transforms.Compose([
                               torchvision.transforms.ToTensor(),
                               torchvision.transforms.Normalize(
                                 (0.1307,), (0.3081,))
                             ])),
  batch_size=batch_size_train, shuffle=True)

model = mnist_torch_model.Net()

test_losses = []
test_counter = [i*len(train_loader.dataset) for i in range(n_epochs + 1)]


acc = evaluate_model(model, None, False)
for epoch in range(1, n_epochs + 1):
  train(epoch, model)
  acc = evaluate_model(model, None, False)  
#print("===========ACC FP32: %.2f ==============" % acc)

input_shape = (1, 1, 28, 28)
sim = QuantizationSimModel(model, default_output_bw=8, default_param_bw=8, input_shapes=input_shape)
# Quantize the untrained MNIST model
# In[]
sim.compute_encodings(forward_pass_callback=evaluate_model, forward_pass_callback_args=1)

print("=============== Quantized Model ==========================")
acc = evaluate_model(sim.model, None, False)  
print("==========================================================")


print("=============== Train Quantized Model ==========================")
acc = evaluate_model(sim.model, None, False) 
train(epoch, sim.model)
acc = evaluate_model(sim.model, None, False)   
print("==========================================================")


sim.export(path='../pytorch_example/', filename_prefix='quantized_mnist', input_shape=input_shape)
```

**Comment by quic-akhobare** [comment]
@jasonlovescoding - I missed your latest comment and hence did not respond.

To answer your questions
1. If you intend to save a QuantizationSim model and later load it back to continue fine-tuning (for example), please use the save_checkpoint and load_checkpoint APIs in aimet_torch.quantsim

2. The exported model from AIMET, which you got from sim.export() call, is a FP32 model that has been optimized to run on quantized runtimes. You can take this model and execute on any runtime. AIMET is not needed to be installed on the runtime itself.

Let me know if these answered your questions.

**Comment by LLNLanLeN** [comment]
I just have some follow up questions, let say I have multiple quantized model checkpoints (which has been saved using save_checkpoint API). I then loaded one selective checkpoint back using `load_checkpoint` function. Now I have a quantized model with all of its parameters. How do I export this checkpoint to `onnx, json and .pth` file? Because it seems like it only load the quantized model back, but it does not recreated a new QuantizationSimModel, so if I used `sim.export()` function, it will crash because due to this function does not exist.

In addition to the question above, is it possible to transfer weights and its quantization parameters to a new model (that has not been pretrained or quantize)? For example:

I have 2 models, A & B. While A is fully pretrained and quantized using AIMET, B is a new model that has not been pretrained or quantized. The only difference between these 2 models is that A use Bilinear Upsampling Layers, while B use Nearest Upsampling layers. Is it possible to transfer weights and quantization parameters from A to B? If it is, what is the best way to do it?
